frontend/src/advising/ManageStudents.tsx

```diff
@@ -29,63 +35,151 @@ const StudentListContainer = styled.div`
+  line-height: 21px;
+  margin: 10px;
+  color: red;
+  &:hover {
+    text-decoration: underline;
+  }
+  cursor: pointer;
+`;
+
+const NoMoreStudents = styled.div`
+  font-size: 10px;
+  line-height: 21px;
+  margin: 10px;
+  color: red;
+`;
+
+const EMPTY_STUDENT_LIST: StudentProps[] = [];
 
 interface StudentsListProps {
   searchQuery: string;
 }
 
+interface StudentsAPI {
+  students: StudentProps[];
+  nextPage: number;
+  lastPage: boolean;
+}
+
 interface StudentProps {
-  name: string;
-  nuid: string;
+  username: string;
+  // nuid: string;
+  email: string;
 }
 
 const ManageStudentsComponent: React.FC = (props: any) => {
   const [searchQuery, setSearchQuery] = useState("");
+
   return (
     <Container>
-      <Search placeholder="Search by name or nuid" onChange={setSearchQuery} />
+      <Search
+        placeholder="Search by name, email, or NUID"
+        onEnter={setSearchQuery}
+      />
       <StudentsList searchQuery={searchQuery} />
     </Container>
   );
 };
 
 const StudentsList = (props: StudentsListProps) => {
+  const [students, setStudents] = useState(EMPTY_STUDENT_LIST);
+  const [isLoading, setIsLoading] = useState(true);
+  const [pageNumber, setPageNumber] = useState(0);
+  const [isLastPage, setIsLastPage] = useState(false);
+  let token = useSelector((state: AppState) => getTokenFromState(state));
+
+  const fetchStudents = (currentStudents: StudentProps[], page: number) => {
+    token = token ? token : "";
+    setIsLoading(true);
+    getStudents(props.searchQuery, page, token)
+      .then((studentsAPI: StudentsAPI) => {
+        setStudents(currentStudents.concat(studentsAPI.students));
+        setPageNumber(studentsAPI.nextPage);
+        setIsLastPage(studentsAPI.lastPage);
+        setIsLoading(false);
+      })
+      .catch(err => console.log(err));
+  };
+
+  useEffect(() => {
+    setStudents(EMPTY_STUDENT_LIST);
+    fetchStudents(EMPTY_STUDENT_LIST, 0);
+  }, [props.searchQuery, token]);
+
   return (
     <StudentListContainer>
+      {isLoading ? (
+        <Loading>
+          <LinearProgress color="secondary" />
+        </Loading>
+      ) : null}
       <StudentListScrollContainer>
-        {LIST_OF_STUDENTS.filter(
-          student =>
-            student.name.includes(props.searchQuery) ||
-            student.nuid.includes(props.searchQuery)
-        ).map(student => (
-          <Student name={student.name} nuid={student.nuid} />
-        ))}
+        {(students === null || students.length == 0) && !isLoading ? (
+          <EmptyState> No students found </EmptyState>
+        ) : (
+          students.map(student => (
+            <Student
+              username={student.username}
+              // nuid={student.nuid}
+              email={student.email}
+              key={student.username + Math.random()}
+            />
+          ))
+        )}
+        {!isLoading ? (
+          isLastPage ? (
+            <NoMoreStudents>No more students</NoMoreStudents>
```

Comment:
Maybe instead of displaying this, we could just not show the "Load More" button

Reply:
I wanted to let the user know that we got no results back. If the user hits Load more and then the button just disappears but nothing came back, it's more ambiguous what happened. Either way - seems like we might want to do infinite scroll here so we can experiment around and see what's best.